File: OGLEngine/SpinningGeometryModel.m
//
//  SpinningGeometryModel.m
//  jj
//
//  Created by Maciej Chmielewski on 17.02.2016.
//  Copyright © 2016 AppUnite. All rights reserved.
//

#import "SpinningGeometryModel.h"

@interface SpinningGeometryModel ()

@property (nonatomic, assign) GLKMatrix4 translation;

@end

@implementation SpinningGeometryModel

- (instancetype)initWithPosition:(GLKVector3)position {
    self = [super init];
    if (self) {
        self.translation = GLKMatrix4MakeTranslation(position.x, position.y, position.z);
    }
    return self;
}

- (GLKMatrix4)modelMatrix {
    return GLKMatrix4Rotate(self.translation, CACurrentMediaTime(), 0.5, 1, 0.25);
}

@end

File: OGLEngine/SpinningGeometryModel.swift
//
//  SpinningGeometryModel.swift
//  OGLEngine
//
//  Created by Maciej Chmielewski on 21.03.2016.
//  Copyright © 2016 MaciejCh. All rights reserved.
//

import Foundation
import GLKit

class SpinningGeometryModel: GeometryModel {
    
    override var orientation: GLKVector3 {
        get {
            let time = Float(CACurrentMediaTime())
            return GLKVector3Make(time, 0, 0)
        }
        set {
            
        }
    }
    
}